src/device.hpp

```diff
@@ -205,7 +205,12 @@ struct cvk_device : public _cl_device_id,
         heap_index = m_mem_properties.memoryTypes[type_index].heapIndex;
         size = std::min(size, m_mem_properties.memoryHeaps[heap_index].size);
 
-        return size;
+        double percentage_of_available_memory =
+            static_cast<double>(config.percentage_of_available_memory) / 100;
+        cvk_info("Using %lf% of total memory size",
+                 config.percentage_of_available_memory);
+
+        return size * percentage_of_available_memory;
```

Comment:
I moved the percentage here as I think `max_alloc_size()` should also address that percentage change.

Reply:
Yes, that's a better location for it.